sql-plugin/src/main/scala-2.12/com/nvidia/spark/rapids/GpuBaseAggregateHelper.scala

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids
+
+import org.apache.spark.sql.catalyst.expressions.aggregate._
+
+object GpuBaseAggregateHelper {
+  def getAggPatternsCanReplace(strPatternToReplace: String) = {
```

Comment:
any reason this doesn't have a return value specified? This code looks the same as the 2.13 version other then that, am I missing it?

Reply:
It's a Scala 2.13 compiler requirement. Let me add a comment to clarify this.

Reply:
I think what @tgravescs is wondering about is: if we just added a return value to `getAppPaternsCanReplace` do you even need to branch this function as two separate impls for 2.12 and 2.13? Or could it be one?

Reply:
I tried that and 2.12 complains about it. The return types are actually different between the 2 versions of Scala

Reply:
It's just in the code that I let the 2.12 version return the inferred type. I forced the type in 2.13 (which I can't force in 2.12).

Reply:
this doesn't make sense to me, standard is to put a return type on a def so that its obvious what its returning. According to my intellij return type here for 2.12 is:

Array[Set[AggregateMode with Product with Serializable]]

So I guess that is different then the 2.13 on that wants  Array[Set[AggregateMode]]

I wonder if you did an asInstanceOf[Array[Set[AggregateMode]]]  but I would have to look more at why scala seems to be doing that.

Reply:
Trying the `asInstanceOf` approach, and it looks promising. Will update if I can remove the split.

Reply:
Looks like `asInstanceOf[Array[Set[AggregateMode]]]` works for both 2.12 and 2.13. I can remove the split now. Will push an update with a comment to note the reason for the asInstanceOf.

Reply:
pushed fix.